### ISSUE TITLE
Sklearn Deprecation Warnings for versions 0.23 and 0.24

### ISSUE THREAD

**Issue by charlesdrotar** [alkaline-ml/pmdarima#252]
**Describe the bug**

After the most recent upgrade of sklearn it has introduced the following deprecation warnings. They really don't need to be resolved until the version of `sklearn` is upgraded to where the changes are breaking. 

- 0.23
  - [x] `check_is_fitted`
    ```
    /Users/runner/hostedtoolcache/Python/3.6.9/x64/lib/python3.6/site-packages/sklearn/utils/validation.py:933: FutureWarning: Passing attributes to check_is_fitted is deprecated and will be removed in 0.23.
    ```

- 0.24
  - [ ] `safe_indexing`
     ```
     /Users/runner/hostedtoolcache/Python/3.6.9/x64/lib/python3.6/site-packages/sklearn/utils/deprecation.py:87: FutureWarning: Function safe_indexing is deprecated; 
     safe_indexing is deprecated in version 0.22 and will be removed in version 0.24.
     ``` 

  - [x] `sklearn.metrics.regression` 
     ```
     FutureWarning: The sklearn.metrics.regression module is  deprecated in version 0.22 and will be removed in version 0.24. The corresponding classes / functions should instead be imported from sklearn.metrics. Anything that cannot be imported from sklearn.metrics is now part of the private API.
     ```

**To Reproduce**  

N/A

**Versions**
`pmdarima` = `1.4.0`
`sklearn` < `0.23`

**Expected behavior**
No warnings triggered.

**Actual behavior**
Warnings are being triggered.

**Additional context**
Keeping this here more for housekeeping so that as `0.23` and `0.24` are released we can proactively anticipate these potentially breaking changes.

**Comment by aaronreidsmith** [comment]
Added `sklearn.metrics.regression` to 0.24 as well

**Comment by tgsmith61591** [comment]
I'm pretty sure this was fixed. We can reopen if we find any more.
